mindsdb/integrations/handlers/merlion_handler/merlion_handler.py

```diff
@@ -0,0 +1,193 @@
+
+    KWARGS_DF = "df"
+
+    DEFAULT_MODEL_TYPE = "default"
+    DEFAULT_MAX_PREDICT_STEP = 100
+    DEFAULT_PREDICT_BASE_WINDOW = 10
+
+    PERSISIT_MODEL_FILE_NAME = "merlion_model"
+    PERSISIT_ARGS_KEY_IN_JSON_STORAGE = "args"
+
+    def create(self, target, args=None, **kwargs):
+        df: pd.DataFrame = kwargs.get(self.KWARGS_DF, None)
+        # prepare arguments
+        task = args.get(self.ARG_USING_TASK, TaskType.forecast.name)
+        model_type = args.get(self.ARG_USING_MODEL_TYPE, self.DEFAULT_MODEL_TYPE)
+        time_column = args.get(self.ARG_USING_TIME_COLUMN, None)
+        horizon = args.get("horizon", self.DEFAULT_MAX_PREDICT_STEP)
+        window = args.get(self.ARG_WINDOW, self.DEFAULT_PREDICT_BASE_WINDOW)
+        # update args for default value maybe has been used, only time column will be set afterwards
+        args[self.ARG_TARGET] = target
+        args[self.ARG_USING_TASK] = task
+        args[self.ARG_USING_MODEL_TYPE] = model_type
+        args["horizon"] = horizon
+        args[self.ARG_WINDOW] = window
+
+        # check df
+        if df is None:
+            raise Exception("missing required key in args: " + self.KWARGS_DF)
+        else:
+            column_sequence = list(df.columns.values)
+            sorted(column_sequence)
+            df = df[column_sequence]
+            args[self.ARG_COLUMN_SEQUENCE] = column_sequence
+
+        # check task, model_type and get the adapter_class
+        adapter_class = self.__args_to_adapter_class(task=task, model_type=model_type)
+        task_enum = TaskType[task]
+
+        # check and cast to ts dataframe
+        time_column = to_ts_dataframe(df=df, time_col=time_column)
+        args[self.ARG_USING_TIME_COLUMN] = time_column
+
+        # train model
+        model_args = {}
+        if task_enum == TaskType.forecast:
+            model_args[MerlionArguments.max_forecast_steps.value] = horizon
+        adapter: BaseMerlionForecastAdapter = adapter_class(**model_args)
+        log.info("Training model, args: " + json.dumps(args))
+        adapter.train(df=df, target=target)
+        log.info("Training model completed.")
+
+        # persist save model
+        model_bytes = adapter.to_bytes()
+        self.model_storage.file_set(self.PERSISIT_MODEL_FILE_NAME, model_bytes)
+        self.model_storage.json_set(self.PERSISIT_ARGS_KEY_IN_JSON_STORAGE, args)
+        log.info("Model and args saved.")
+
+    def predict(self, df):
+        rt_df = df.copy(deep=True)
+        # read model and args from storage
+        model_bytes = self.model_storage.file_get(self.PERSISIT_MODEL_FILE_NAME)
+        args = self.model_storage.json_get(self.PERSISIT_ARGS_KEY_IN_JSON_STORAGE)
+
+        # resolve args
+        task = args[self.ARG_USING_TASK]
+        model_type = args[self.ARG_USING_MODEL_TYPE]
+        time_column = args[self.ARG_USING_TIME_COLUMN]
+        target = args[self.ARG_TARGET]
+        horizon = args["horizon"]
+        # window = args[self.ARG_WINDOW]
```

Comment:
If `window` is not needed, then you can remove this comment 👍

Reply:
fixed